python/lsst/dax/obscore/siav2.py

```diff
@@ -0,0 +1,703 @@
+            One or more query that is instrument-specific. These will be
+            treated as independent queries combined with the general query
+            constraints.
+        general_wheres : `WhereBind` | None
+            Query constraint that is not instrument-specific.
+        """
+        instrument_wheres = []
+        general_where: WhereBind | None = None
+        if "instrument" in dimensions and instruments:
+            # Need separate where clauses for each instrument if
+            # we also are using a physical filters constraint.
+            if "physical_filter" in dimensions and "filters" in band_info:
+                for i, inst in enumerate(instruments):
+                    instrument_wheres.append(
+                        WhereBind(
+                            where=f"instrument = INST{i} AND physical_filter in (phys)",
+                            bind={"phys": band_info["filters"][inst], f"INST{i}": inst},
+                        )
+                    )
+            else:
+                # Can include all instruments in single query.
+                instrument_wheres.append(
+                    WhereBind(where="instrument IN (INSTRUMENTS)", bind={"INSTRUMENTS": instruments})
+                )
+        elif "band" in dimensions and "bands" in band_info:
+            # Band is not needed for an instrument query since
+            # we will be using physical filters for those.
+            general_where = WhereBind(where="band IN (bands)", bind={"bands": band_info["bands"]})
+        return instrument_wheres, general_where
+
+    def from_pos(self, regions: Sequence[Region], dimensions: DimensionGroup) -> WhereBind | None:
+        """Convert a region request to a butler where clause.
+
+        Parameters
+        ----------
+        regions : `~collections.abc.Iterable` [ `lsst.sphgeom.Region` ]
+            The region of interest.
+        dimensions : `lsst.daf.butler.DimensionGroup`
+            The dimensions for the dataset type being queried.
+
+        Returns
+        -------
+        where : `WhereBind` or `None`
+            The where clause or `None` if region is not supported by this
+            dataset type.
+        """
+        extra_dims = set()
+        region_dim = dimensions.region_dimension
+        if not region_dim:
+            if "exposure" in dimensions:
+                # Special case for Rubin default universe.
+                region_dim = "visit_detector_region"
+                extra_dims = {"visit"}
+            else:
+                return None
+
+        if not regions:
+            # In case we are called with no regions.
+            return WhereBind(where="")
+
+        # Must OR together all regions. We can do this in sphgeom directly.
+        # Butler query system itself can not OR regions.
+        region = regions[0]
+        if len(regions) > 1:
+            for r in regions[1:]:
+                region = UnionRegion(region, r)
```

Comment:
Would be nice if `UnionRegion` could take any number of regions in its constructor.

Reply:
Indeed it would.